controllers/control/k8ssandratask_controller.go

```diff
@@ -0,0 +1,212 @@
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	api "github.com/k8ssandra/k8ssandra-operator/apis/control/v1alpha1"
+)
+
+// K8ssandraTaskReconciler reconciles a K8ssandraTask object
+type K8ssandraTaskReconciler struct {
+	*config.ReconcilerConfig
+	client.Client
+	Scheme      *runtime.Scheme
+	ClientCache *clientcache.ClientCache
+}
+
+//+kubebuilder:rbac:groups=control.k8ssandra.io,namespace="k8ssandra",resources=k8ssandratasks,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=control.k8ssandra.io,namespace="k8ssandra",resources=k8ssandratasks/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=control.k8ssandra.io,namespace="k8ssandra",resources=k8ssandratasks/finalizers,verbs=update
+
+func (r *K8ssandraTaskReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx).WithValues("K8ssandraTask", req.NamespacedName)
+
+	logger.Info("Fetching task", "K8ssandraTask", req.NamespacedName)
+	k8Task := &api.K8ssandraTask{}
+	err := r.Get(ctx, req.NamespacedName, k8Task)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{RequeueAfter: r.ReconcilerConfig.DefaultDelay}, err
+	}
+	// TODO k8Task.DeepCopy()? why?
+
+	if k8Task.DeletionTimestamp != nil {
+		// TODO delete dependent objects?
+		return ctrl.Result{}, nil
+	}
+
+	kcKey := k8Task.GetClusterKey()
+	logger.Info("Fetching cluster", "K8ssandraCluster", kcKey)
+	kc := &k8sapi.K8ssandraCluster{}
+	err = r.Get(ctx, kcKey, kc)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	dcs, err := filterDcs(kc, k8Task.Spec.Datacenters)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	for _, dc := range dcs {
+		dcNamespace := utils.FirstNonEmptyString(dc.Meta.Namespace, kc.Namespace)
+
+		desiredDcTask := newDcTask(k8Task, dcNamespace, dc.Meta.Name)
+
+		remoteClient, err := r.ClientCache.GetRemoteClient(dc.K8sContext)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		dcTaskKey := client.ObjectKey{
+			Namespace: desiredDcTask.Namespace,
+			Name:      desiredDcTask.Name,
+		}
+		actualDcTask := &cassapi.CassandraTask{}
+		err = remoteClient.Get(ctx, dcTaskKey, actualDcTask)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				logger.Info("Creating DC task", "CassandraTask", dcTaskKey)
+				err = remoteClient.Create(ctx, desiredDcTask)
+				if err != nil {
+					return ctrl.Result{}, err
+				}
+			} else {
+				return ctrl.Result{}, err
+			}
+		} else {
+			if k8Task.Status.Datacenters == nil {
+				k8Task.Status.Datacenters = make(map[string]cassapi.CassandraTaskStatus)
+			}
+			k8Task.Status.Datacenters[dc.Meta.Name] = actualDcTask.Status
+		}
+
+		k8Task.BuildGlobalStatus()
+		err = r.Status().Update(ctx, k8Task)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	return ctrl.Result{}, nil
```

Comment:
As indicated in the previous comment - if all the intended tasks are not finished, you need to requeue here.